pkg/registries/docker/metadata_v1.go

```diff
@@ -127,48 +120,41 @@ func (r *Registry) handleV1ManifestLayer(remote string, ref digest.Digest) (*sto
 	if err != nil {
 		return nil, err
 	}
-	img := &image.Image{}
+	img := &v1.Image{}
 	if err := json.Unmarshal(val, img); err != nil {
 		return nil, err
 	}
 
 	var layers []*storage.ImageLayer
 	for _, h := range img.History {
-		// See github.com/moby/moby/image/image.go
 		instruction, value := lineToInstructionAndValue(h.CreatedBy)
 		layers = append(layers, &storage.ImageLayer{
-			Created:     protoconv.ConvertTimeToTimestamp(h.Created),
-			Author:      h.Author,
+			Created:     protoconv.ConvertTimeToTimestampOrNow(h.Created),
+			Author:      h.CreatedBy,
 			Instruction: instruction,
 			Value:       value,
 			Empty:       h.EmptyLayer,
 		})
 	}
 
-	var labels map[string]string
-	if img.Config != nil {
-		labels = img.Config.Labels
-	}
-
 	var metadata = &storage.V1Metadata{
 		Digest:  ref.String(),
-		Created: protoconv.ConvertTimeToTimestamp(img.Created),
-		Labels:  labels,
+		Created: protoconv.ConvertTimeToTimestampOrNow(img.Created),
+		Labels:  img.Config.Labels,
 	}
 
-	if img.Config != nil {
-		metadata.Volumes = make([]string, 0, len(img.Config.Volumes))
-		for k := range img.Config.Volumes {
-			metadata.Volumes = append(metadata.Volumes, k)
-		}
+	metadata.Volumes = make([]string, 0, len(img.Config.Volumes))
+	for k := range img.Config.Volumes {
+		metadata.Volumes = append(metadata.Volumes, k)
+	}
 
-		metadata.User = "root"
-		if img.Config.User != "" {
-			metadata.User = img.Config.User
-		}
-		metadata.Command = img.Config.Cmd
-		metadata.Entrypoint = img.Config.Entrypoint
+	metadata.User = "root"
+	if img.Config.User != "" {
```

Comment:
`metadata.User` defaults to `""` anyway, right? I don't think there is a need to check for `""` here, right?

Reply:
not important for this backport, but may be nice to update in `master`

Reply:
agree, but this is the old code. I've tried to make diff minimal. here we only need to remove indent.